src/Http/ResponseEmitter.php

```diff
@@ -264,16 +269,28 @@ protected function emitCookies(array $cookies): void
             if (is_string($data['expires'])) {
                 $data['expires'] = strtotime($data['expires']);
             }
-            /** @psalm-suppress PossiblyInvalidArgument */
-            setcookie(
-                $data['name'],
-                $data['value'],
-                $data['expires'],
-                $data['path'],
-                $data['domain'],
-                $data['secure'],
-                $data['httponly']
-            );
+
+            $name = $data['name'];
+            $value = $data['value'];
+            unset($data['name'], $data['value']);
+            if (!$data['samesite']) {
+                unset($data['samesite']);
+            }
+
+            if (PHP_VERSION_ID >= 70300) {
+                setcookie($name, $value, $cookie['options']);
```

Comment:
Should we raise an exception if people try and use samesite cookies when they are on PHP<7.3?

Reply:
I'll add an exception.

Reply:
Could it generate header with `Cookie::toHeaderValue()` and send it using `header()` instead?

Exceptions might be missed if development and production uses different PHP version.

Reply:
I guess that's an option too.

Reply:
Directly switching to `header()` for PHP < 7.3 would making testing a pain and we would need different set of tests for different PHP versions.

Instead we can use the namespaced `Cake\Http\setcookie()` currently only used in tests in `ResponseEmitter` too which can internally either call PHP's `setcookie()` or `header()`.

Reply:
I would use same `header()` always independently of PHP.